packages/integrations/vercel/src/serverless/middleware.ts

```diff
@@ -68,12 +65,12 @@ export default async function middleware(request, context) {
 	});
 	ctx.locals = ${handlerTemplateCall};
 	const next = async () => {
-		const response = await fetch(url, {
+		return new Response(null, {
 			headers: {
-				${JSON.stringify(ASTRO_LOCALS_HEADER)}: trySerializeLocals(ctx.locals)
+				'x-middleware-next': '1',
+				'x-astro-locals': trySerializeLocals(ctx.locals)
 			}
-		});
-		return response;
+		})
```

Comment:
Need to make this work without returning an empty response somehow. Maybe it could fetch an internal-only route that leaves out `middlewarePath`.